### DISCUSSION TITLE
Midi out for Jukebox mode

### DISCUSSION THREAD

**Discussion by streetuff** [copych/AcidBox#10] [Ideas]
Hi!

would be usefull, if a midi out mode could be implemented for the jukebox mode. That way i could control external devices like Roland MC-303, Behringer TD-8, Korg Volca NuBass, Roland TR6-S. Maybe make it switchable via a GPIO. External devices should probably only receive Note data and no CC-Messages, so you could tweak the potentiometers youself :)

**Comment by copych** [comment]
all that's possible. for now there's a #define MIDI_RAMPS in config.h which turns off all CC automation when commented out. and you can attach potentiometers to the defined pins. There're 3 pots configured to play with the second 303 cutoff, reso and distortion. Midi output shouldn't be hard to code as well.

**Reply by streetuff** [reply]
Yeah. Have seen your recent update. But i dont want to use POTs on the ESP, but on my synths :)
Could be super fun, to play the acidbanger on external synths.

**Reply by copych** [reply]
I don't have either ESP32 or some synth by me right now, can you check if the next modification will result as expected:
Starting at line 297 of AcidBanger.ino, replace the two following functions with this :
```
static void send_midi_noteon(byte chan, byte note, byte vol) {
#ifdef MIDI_VIA_SERIAL  
  MIDI.sendNoteOn(note, vol, chan);
#endif
#ifdef MIDI_VIA_SERIAL2  
  MIDI2.sendNoteOn(note, vol, chan);
#endif
  handleNoteOn( chan, note, vol) ;
}

static void send_midi_noteoff(byte chan, byte note) {
#ifdef MIDI_VIA_SERIAL  
  MIDI.sendNoteOn(note, 0, chan);
#endif
#ifdef MIDI_VIA_SERIAL2  
  MIDI2.sendNoteOn(note, 0, chan);
#endif
  handleNoteOff( chan, note, 0) ;
}
```

**Reply by streetuff** [reply]
I can confirm, that it is working. just tried with the behringer TD-3, Volca NuBass and the Roland TR6-S. Very cool!

**Reply by 01GOD** [reply]
Nice. That is cool.

**Reply by smutax** [reply]
> I don't have either ESP32 or some synth by me right now, can you check if the next modification will result as expected: Starting at line 297 of AcidBanger.ino, replace the two following functions with this :
> 
> ```
> static void send_midi_noteon(byte chan, byte note, byte vol) {
> #ifdef MIDI_VIA_SERIAL  
>   MIDI.sendNoteOn(note, vol, chan);
> #endif
> #ifdef MIDI_VIA_SERIAL2  
>   MIDI2.sendNoteOn(note, vol, chan);
> #endif
>   handleNoteOn( chan, note, vol) ;
> }
> 
> static void send_midi_noteoff(byte chan, byte note) {
> #ifdef MIDI_VIA_SERIAL  
>   MIDI.sendNoteOn(note, 0, chan);
> #endif
> #ifdef MIDI_VIA_SERIAL2  
>   MIDI2.sendNoteOn(note, 0, chan);
> #endif
>   handleNoteOff( chan, note, 0) ;
> }
> ```

Hi there. Is it correct that above had been committed (5b9e5b9) and doesn't need adding to AcidBanger.ino any longer? Thanks

**Comment by copych** [comment]
Yes, noteOn and noteOff are sent to the active midi ports. Just check config.h to select appropriate port(s).
